core/eolearn/core/utils/testing.py

```diff
@@ -77,6 +77,9 @@ def _generate_feature_data(
 
 def _get_feature_shape(
     rng: np.random.Generator, ftype: FeatureType, timestamps: List[dt.datetime], config: PatchGeneratorConfig
-) -> Tuple[int, ...]:
-    time, height, width, depth = len(timestamps), *config.raster_shape, rng.integers(*config.depth_range)
-    return (time, height, width, depth) if ftype.is_temporal() else (height, width, depth)
+) -> List[int]:
+    shape = [len(timestamps)] if ftype.is_temporal() else []  # time
+    f_dim: int = ftype.ndim() or 0
+    shape.extend(config.raster_shape if f_dim >= 3 else [])  # height, width
+    shape.append(rng.integers(*config.depth_range))  # depth
+    return shape
```

Comment:
I cannot understand this code at all
Describe the problem & solution to me in human words.

Reply:
The issue here is that the old code assumes that the feature shape will always be of image type, like discussed internally in https://git.sinergise.com/eo/code/eo-learn/-/issues/194

We had this piece of code before we (@zigaLuksic and me) together changed it to what is seen on the left.

Basically this constructs the feature shape based on the feature type. Not sure why L82 is there though. @jgersak if you were afraid that this would fail in case of meta info or vector types, this shouldn't be the case, because only raster type features are allowed, see L55

Reply:
```py
    time, height, width, depth = len(timestamps), *config.raster_shape, rng.integers(*config.depth_range)
    shape_dict = {4: (time, height, width, depth), 3: (height, width, depth), 2: (time, depth), 1: (depth,)}
    return shape_dict[ftype.ndim()]
```
this was first solution and matic thought that it is too much of "black magic" so i do adaptation of his:

current solution should first look if ftype is_temporal and int len timestamps  or empty list dependent on that 
second test is if ftype.ndim() > 2 all our features that are that dim have spatial component so raster_shape can be append 
all features have free component on last place (L84)

Reply:
L82  is to make mypy happy xD

Reply:
> this was first solution and matic thought that it is too much of "black magic" so i do adaptation of his:


It looks more elegant, but I just wasn't sure if it's safe to assume that a single dimension number always corresponds to a unique shape representation. currently this seems to be the case, but ... 

perhaps I'm wrong and this is better, I thought that the changes here were obvious enough

Reply:
How about we focus on code that is human readable, like the original was?
The first thing that i noticed is that yall were saying "not eveything is an image"
good, try something like this then.

```python
    time, depth = len(timestamps), rng.integers(*config.depth_range)
    if <is_image_ftype>:
        return (time, height, width, depth) if ftype.is_temporal() else (height, width, depth)
    return <whatever belongs here>
```

If things are not images we also have no clue what the dimension is supposed to be anyway (or at leas the current config doesn't tell us) so such a separation is much more sensible.

Reply:
reverted,  update supported_feature_types in patch_generator